packages/image_picker/image_picker_platform_interface/lib/src/types/media_options.dart

```diff
@@ -13,11 +13,47 @@ class MediaOptions {
   const MediaOptions({
     this.imageOptions = const ImageOptions(),
     required this.allowMultiple,
+    this.limit,
   });
 
+  /// Construct a new MediaOptions instance.
+  ///
+  /// Throws if limit is lower than 2,
+  /// or allowMultiple is false and limit is not null
+  MediaOptions.createAndValidate({
+    this.imageOptions = const ImageOptions(),
+    required this.allowMultiple,
+    this.limit,
+  }) {
+    _validate(allowMultiple: allowMultiple, limit: limit);
+  }
+
   /// Options that will apply to images upon selection.
   final ImageOptions imageOptions;
 
   /// Whether to allow for selecting multiple media.
   final bool allowMultiple;
+
+  /// The maximum number of images to select.
+  ///
+  /// Default null value means no limit.
```

Comment:
We should explicitly document that this value may be ignored by platforms that cannot support it.